### DISCUSSION TITLE
Needed Task Scheduling (Cron Job)

### DISCUSSION THREAD

**Discussion by DannyelPina** [adonisjs/core#4617] [Ideas]
Thanks for providing that amazing js/ts framework. I love that and enjoy working with it in my daily life.

One thing I missed here, is the ability to schedule tasks, for specific times, daily or even monthly.

Like this one from Laravel link: https://laravel.com/docs/11.x/scheduling

**Comment by svenhoerig** [comment]
For AdonisJS Version 5 I used [adonis5 scheduler from reg2005](https://github.com/reg2005/adonis5-scheduler) but it's not compatible with AdonisJS 6 yet. For AdonisJS Version 6 I now use [adonisjs-scheduler from GABBOUCHI](https://github.com/KABBOUCHI/adonisjs-scheduler)

**Reply by bytelucas** [reply]
Can you help me when I config adonis5-scheduler

`➜  hcomm-smartdesk-api-v2 git:(main) ✗ yarn add adonis5-scheduler         
yarn add v1.22.22
warning ../../../../../package.json: No license field
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "@japa/preset-adonis > @japa/assert > api-contract-validator > api-schema-builder > swagger-parser > @apidevtools/swagger-parser@10.0.3" has unmet peer dependency "openapi-types@>=7".
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
success Saved 12 new dependencies.
info Direct dependencies
└─ adonis5-scheduler@2.1.1
info All dependencies
├─ @adonisjs/generic-exceptions@3.0.1
├─ adonis5-scheduler@2.1.1
├─ cron-parser@4.9.0
├─ cron-time-generator@2.0.3
├─ cronstrue@2.50.0
├─ long-timeout@0.1.1
├─ node-exceptions@4.0.1
├─ node-schedule@2.1.1
├─ proper-lockfile@4.1.2
├─ retry@0.12.0
├─ sorted-array-functions@1.3.0
└─ upcast@4.0.0
✨  Done in 3.17s.
➜  hcomm-smartdesk-api-v2 git:(main) ✗ node ace invoke adonis5-scheduler

UPDATE: tsconfig.json { types += "adonis5-scheduler" }
UPDATE: .adonisrc.json { commands += "adonis5-scheduler/build/commands" }
UPDATE: .adonisrc.json { providers += "adonis5-scheduler" }
[ warn ]  Unable to generate manifest file. Check the following error for more info

   Error:
require()
of
ES
Module
/Users/hieu/Documents/Companies/H-ecom/source-code/hcomm-smartdesk-api-v2/node_modules/@adonisjs/ace/build/index.js
from
/Users/hieu/Documents/Companies/H-ecom/source-code/hcomm-smartdesk-api-v2/node_modules/adonis5-scheduler/build/commands/MakeTask.js
not
supported.

Instead
change
the
require
of
index.js
in
/Users/hieu/Documents/Companies/H-ecom/source-code/hcomm-smartdesk-api-v2/node_modules/adonis5-scheduler/build/commands/MakeTask.js
to
a
dynamic
import()
which
is
available
in
all
CommonJS
modules.
 

   ⁃ Object.<anonymous>
     /Users/hieu/Documents/Companies/H-ecom/source-code/hcomm-smartdesk-api-v2/node_modules/adonis5-scheduler/build/commands/MakeTask.js:12`
